jsEx/Form.js

```diff
@@ -0,0 +1,106 @@
+function validate() {
+  let username = document.getElementById("username").value;
+  let password = document.getElementById("password").value;
+  let confirmPassword = document.getElementById("confirm-password").value;
+
+  let isValidUser = isValidUsername(username);
+  let isValidPass = isValidPassword(password);
+  let isValidConfirmPass = isValidConfirmPassword(confirmPassword, password);
+
+  if (isValidUser) {
+    hideErrorMessag("invalidUsername");
+  }
+  if (isValidPass) {
+    hideErrorMessag("invalidPassword");
+  }
+  if (isValidConfirmPass) {
+    hideErrorMessag("invalidConfirmPassword");
+  }
+
+  return (isValidUser && isValidPass && isValidConfirmPass) ? request() : false;
+}
+
+function ajax(dataRequest) {
+  var xhr = new XMLHttpRequest();
+
+  xhr.open(dataRequest.method, dataRequest.url, true);
+
+  xhr.onreadystatechange = function() {
+    if (xhr.readyState === XMLHttpRequest.DONE && xhr.status === 200) {
+      dataRequest.success(xhr.response);
+    } else {
+      console.error(xhr.response);
```

Comment:
You may want to notify the user in some way about this exceptional case, e.g. text on the page somewhere.

Reply:
Also, currently the way this code behaves makes the console print text in red as error, while the request has been successful. This should not be happening on successful request.
![image](https://user-images.githubusercontent.com/10218337/78072634-1f985b80-73a8-11ea-9a6c-cb5dbe7de16f.png)